client/send.go

```diff
@@ -0,0 +1,131 @@
+// eventsgateway
+// https://github.com/topfreegames/eventsgateway
+//
+// Licensed under the MIT license:
+// http://www.opensource.org/licenses/mit-license
+// Copyright © 2018 Top Free Games <backend@tfgco.com>
+
+package client
+
+import (
+	"math"
+	"time"
+
+	uuid "github.com/satori/go.uuid"
+	"github.com/sirupsen/logrus"
+	log "github.com/sirupsen/logrus"
+	pb "github.com/topfreegames/protos/eventsgateway/grpc/generated"
+	context "golang.org/x/net/context"
+)
+
+// Send sends an event to another server via grpc using the client's configured topic
+func (g *GRPCClient) Send(name string, props map[string]string) error {
+	l := g.logger.WithFields(log.Fields{
+		"operation": "send",
+		"event":     name,
+	})
+	l.Debug("sending event")
+	if err := g.sendEvent(name, g.topic, props); err != nil {
+		l.WithError(err).Error("send event failed")
+		return err
+	}
+	return nil
+}
+
+// SendToTopic sends an event to another server via grpc using an explicit topic
+func (g *GRPCClient) SendToTopic(name, topic string, props map[string]string) error {
+	l := g.logger.WithFields(log.Fields{
+		"operation": "sendToTopic",
+		"event":     name,
+		"topic":     topic,
+	})
+	l.Debug("sending event")
+	if err := g.sendEvent(name, topic, props); err != nil {
+		l.WithError(err).Error("send event failed")
+		return err
+	}
+	return nil
+}
+
+func (g *GRPCClient) sendEvent(name, topic string, props map[string]string) error {
+	g.logger.WithFields(log.Fields{
+		"operation": "eventRequest",
+		"name":      name,
+	}).Debug("getting event request")
+
+	event := &pb.Event{
+		Id:        uuid.NewV4().String(),
+		Name:      name,
+		Topic:     topic,
+		Props:     props,
+		Timestamp: time.Now().UnixNano() / int64(time.Millisecond),
+	}
+
+	if g.async {
+		g.eventsChannel <- event
+		return nil
+	} else {
+		ctx, _ := context.WithTimeout(context.Background(), g.timeout)
```

Comment:
Instead of creating a new context, it would be better to propagate the existing one to here. Same throughout.

Reply:
client/sync now does just that. client/async continues using context.Background() since calls don't happen in the same context.